### ISSUE TITLE
SEI: Adapter class needed for data cleaning

### ISSUE THREAD

**Issue by FFunk** [FFunk/vegetationClassification#2]
Problem: The sensitive ecosystem inventory project contains dirty data and requires the removal, redistribution of major and minor classes, as well as the addition of new classes.

Issue log:
- All: Many subclasses do not contain enough samples and have to be omitted from training
- Alpine: Alpine ecosystems are not relevant for the floodplain analysis and should be removed.
- Forest: Forest age is not visually accessible and hence young forest, young forest (small), mature forest, old forest are summarized into a single major class, Forest
- Forest:  The forest subclass "mixed forest" is an arbitrary, and mostly a 'lazy' class which is more confusing than helpful. Mixed forest should be relabeled as "undetermined" and substituted by the correct labels (coniferous, broadleaf). The ML algorithm should not learn this class but still, predict on its area.
- Riparian: The Riparian ecosystem has numerous subclasses that require exact timing for labeling (low, high, and medium bench floodplain) or are usually blocked by forest canopy (fringe, gully)
- Riparian: The Riparian subclasses lake, river, and pond require too much context to be differentiated and should be merged into the subclass water
- Riparian: The Riparian ecosystem needs an additional shapefile for gravel bars.
- Sparsely Vegetated: The major class "Sparsely Vegetated" contains the subclass "Talus". This subclass is not identifiable from a top-down view. It is also usually overgrown by forest and woodland and hence causes lots of confusion. The subclass Talus and its corresponding major class should be removed)
- Sparsely vegetated: Relabel rock outcrop to rock.
- Estuarine: Tidalflats are mudflats and should be rather recategorized as intertidal zones.
- Intertidal zones: Eelgrass polygons are largely incorrect. Intertidal zones contain lots of water. Eelgrass labels should be removed as well (with the IT) label.
- Wetlands: Shallow water is a highly colliding class. Most water bodies such as lakes, rivers, etc. contain shallow water. On top, shallow water may or may not be there due to seasonal change. Hence, it should rather be predicted as water.

Proposed solution:
- Add extra samples in preprocessing steps.
- Write an adapter function that takes a (height, width, numb-classes) binary matrix and make the readjustments, resulting in (height, width, adj_numb_classes).
- Maintain vectors for trainable_classes (which takes the classes that should be learned) as well as predict_classes (the class area on which a prediction is possible and should be executed) for minor and major classes. Also, save class labels.
- Accuracies then can be obtained for trainable_classes. Predictions generated on predict_classes

**Comment by FFunk** [comment]
Resolved: Added sei_cleaner.py which facilitates the cleaning. Added changes necessary to the CNN preparation file (all classes)
